crates/core/component/sct/src/component/rpc.rs

```diff
@@ -55,4 +56,24 @@ impl QueryService for Server {
             anchor: anchor.map(Into::into),
         }))
     }
+
+    #[instrument(skip(self, request))]
+    async fn timestamp_by_height(
+        &self,
+        request: tonic::Request<TimestampByHeightRequest>,
+    ) -> Result<tonic::Response<TimestampByHeightResponse>, Status> {
+        let state = self.storage.latest_snapshot();
+
+        let height = request.get_ref().height;
+        let block_time = state.get_block_timestamp().await.map_err(|e| {
```

Comment:
This will always return the timestamp for the latest block in the snapshot

Reply:
I don't think we store height-by-height timestamp, but I can take a look once I'm back at the computer

Reply:
Correct, we would need to add a new nonverifiable storage key containing timestamp-by-height